mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/WorkerExecutionOperationsNetworkStage.java

```diff
@@ -622,7 +627,9 @@ private WorkerConsumer connectToObservableAtPreviousStages(Observable<JobSchedul
 
     @Override
     public void shutdownStage() throws IOException  {
-        logger.info("Shutdown initiated");
+        if (jobStatus != null) {
+            jobStatus.onNext(heartbeatRef.get().getFailedHeartbeatStatus());
```

Comment:
should shutdown always reflect as failed? what about regular shutdown?

Reply:
Good question. I wasn't able to split the case failure vs regular shutdown which frankly would be ideal, however sending this signal to master during regular shutdown won't do anything because the worker has already been archived. The job actor will simply log `Job XYZ Terminal event from Unknown worker ZYX. Ignoring`.